### PR TITLE
CMake Tutorial

### PR DESCRIPTION
**Pull request by brunoalano**
A simple tutorial about how to use CMake instead of plain-old Makefiles and Autoconfigure

### CODE REVIEW

File: cmake.html.markdown [hunk 1]
```diff
@@ -0,0 +1,176 @@
+---
+language: cmake
+contributors:
+    - ["Bruno Alano", "https://github.com/brunoalano"]
+filename: CMake
+---
+
+CMake it's a cross-platform, open-source build system. This tool will allow you
```
Comment (adambard) [inline cmake.html.markdown]: it's -> is

File: cmake.html.markdown [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+to test, compile and create packages of your source code.
+
+The problem that CMake tries to solve it's the problem of Makefiles and
```
Comment (adambard) [inline cmake.html.markdown]: it's -> is

File: cmake.html.markdown [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+Autoconfigure on cross-platform (different make interpreters have different
+command) and the ease-of-use on linking 3rd party libraries.
+
+CMake is an extensible, open-source system that manages the build process in
+an operating system and in a compiler-independent manner. Unlike many
```
Comment (adambard) [inline cmake.html.markdown]: I would say, "CMake is an extensible, open-source system that manages the build process in
an operating-system- and compiler-independent manner."

File: cmake.html.markdown [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+```cmake
+# In CMake, this is a comment
+
+# To run our code, we will use these steps:
+#  - mkdir build && cd build
+#  - cmake ..
+#  - make
+# 
+# With those steps, we will follow the best pratice to compile into a subdir
```
Comment (adambard) [inline cmake.html.markdown]: pratice -> practice

File: cmake.html.markdown [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+# The CMake file MUST be named as "CMakeLists.txt".
+
+# Setup the minimum version required of CMake to generate the Makefile
+cmake_minimum_required (VERSION 2.8)
+
+# Raises a FATAL_ERROR if version < 2.8
+cmake_minimum_required (VERSION 2.8 FATAL_ERROR)
+
+# We setup the name for our project. After we do that, this will change some
+# directories naming convention genearted by CMake. We can send the LANG of
```
Comment (adambard) [inline cmake.html.markdown]: genearted -> generated

File: cmake.html.markdown [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+# code as second param
```
Comment (adambard) [inline cmake.html.markdown]: code -> the code

File: cmake.html.markdown [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+# Include Directories
+# In GCC, this will invoke the "-I" command
+include_directories( include )
+
+# Where are the additional libraries installed? Note: provide includes
+# path here, subsequent checks will resolve everything else
+set( CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/CMake/modules/" )
+
+# Conditions
+if ( CONDITION )
```
Comment (adambard) [inline cmake.html.markdown]: Perhaps an example of what CONDITION could be would be better?

File: cmake.html.markdown [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+
+while(condition)
+  COMMAND1(ARGS ...)
+  COMMAND2(ARGS ...)
+  ...
+endwhile(condition)
+
+
+# Logic Operations
+if(FALSE AND (FALSE OR TRUE))
```
Comment (adambard) [inline cmake.html.markdown]: This would probably suffice as an example for if

File: cmake.html.markdown [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+
+# Link the libraries
+target_link_libraries( ${PROJECT_NAME} ${LIBS} m )
+
+# Where are the additional libraries installed? Note: provide includes
+# path here, subsequent checks will resolve everything else
+set( CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/CMake/modules/" )
+
+# Compiler Condition (gcc ; g++)
+if ( "${CMAKE_C_COMPILER_ID}" STREQUAL "GNU" )
```
Comment (adambard) [inline cmake.html.markdown]: Oh this one is good too. You can just leave out introducing `if` until you need to, people know what `if` is.

File: cmake.html.markdown [hunk 10]
```diff
@@ -0,0 +1,176 @@
+---
+language: cmake
+contributors:
+    - ["Bruno Alano", "https://github.com/brunoalano"]
+filename: CMake
```
Comment (adambard) [inline cmake.html.markdown]: The filename could be `CMakeLists.txt` -- only the code parts are included in the generated file.

File: cmake.html.markdown (hunk 1 above)
Comment (adambard) [inline cmake.html.markdown]: Oh, also if you could move this to the "tools" section by adding the following:

```
category: tool
tool: tmux
```

This should replace the `language` specifier in the heading to ensure it ends up in the right spot.

**Comment by vendethiel** [comment]
@brunoalano hey, any news?

**Comment by vendethiel** [comment]
Going to close this one for now :). Feel free to ping us here if you find some time working on it again! Thanks.

**Pull request closed without merging**